cl/recap/mergers.py

```diff
@@ -409,26 +412,73 @@ def get_order_of_docket(docket_entries):
     return order
 
 
-def make_recap_sequence_number(de):
+def convert_to_court_timezone(
+    datetime_filed: datetime, court_id: str
+) -> datetime:
+    """Convert a docket entry datetime filed to the court timezone it belongs
+    to.
+
+    :param datetime_filed: The docket entry datetime filed
+    :param court_id: The court id to which docket entries belong, used for
+    timezone conversion.
+    :return: A date object in the court timezone.
+    """
+    court_timezone = COURT_TIMEZONES.get(court_id, "US/Eastern")
+    return do_timezone(datetime_filed, court_timezone)
+
+
+def set_docket_entry_date_and_time(
+    de: DocketEntry, date_filed: date | datetime
+) -> None:
+    """Set the docket entry date and time, if datetime is available, convert it
+    to the local court timezone, split it into date and time and save it to
+    date_filed and time_filed.
+
+    :param de: The docket entry to set date and time filed.
+    :date_filed: The date or datetime instance provided by the source.
+    :return: None
+    """
+
+    if isinstance(date_filed, datetime):
+        datetime_filed_local = convert_to_court_timezone(
+            date_filed, de.docket.court.pk
+        )
+        de.time_filed = datetime_filed_local.time()
+        date_filed = datetime_filed_local.date()
+    else:
+        # If not time data is available, compare if date_filed changed if so
+        # restart time_filed to None.
+        if de.date_filed != date_filed:
+            de.time_filed = None
+    de.date_filed = date_filed or de.date_filed
+
+
+def make_recap_sequence_number(de: dict, court_id: str) -> str:
```

Comment:
Looking at the complexity of this function, I wonder if it'd be simpler if it just returned the value rather than setting it on the `de`. Kind of like my earlier comment. That way, you can pass in the date_filed date object, and the recap_sequence_number, and get a useful value as a return and you don't have to mess around with timezone stuff in the function (it's already done when you pass it in).

Reply:
Changed, now the method is simpler and clearer!